app/src/sharedTest/java/org/oppia/android/app/profile/ProfileChooserFragmentTest.kt

```diff
@@ -582,99 +335,82 @@ class ProfileChooserFragmentTest {
     }
   }
 
   @Test
-  // TODO(#973): Fix ProfileChooserFragmentTest
-  @Ignore
   fun testProfileChooserFragment_multipleProfiles_checkText_addProfileIsVisible() {
     profileTestHelper.initializeProfiles()
     launch<ProfileChooserActivity>(createProfileChooserActivityIntent()).use {
-      onView(atPositionOnView(R.id.profile_recycler_view, 3, R.id.add_profile_text))
-        .check(matches(withText(R.string.profile_chooser_add)))
+      testCoroutineDispatchers.runCurrent()
+      matchStringOnProfileListItem(
+        position = 3,
+        targetView = R.id.add_profile_text,
+        stringToMatch = context.getString(R.string.profile_chooser_add)
+      )
     }
   }
 
   @Test
-  // TODO(#973): Fix ProfileChooserFragmentTest
-  @Ignore
   fun testProfileChooserFragment_multipleProfiles_checkDescriptionText_isDisplayed() {
     profileTestHelper.initializeProfiles()
     launch<ProfileChooserActivity>(createProfileChooserActivityIntent()).use {
+      testCoroutineDispatchers.runCurrent()
       onView(
         atPositionOnView(
           R.id.profile_recycler_view,
-          3, R.id.add_profile_description_text
+          3,
+          R.id.add_profile_description_text
         )
       ).check(matches(not(isDisplayed())))
     }
   }
 
-  private fun getCurrentActivity(): Activity? {
-    var currentActivity: Activity? = null
-    getInstrumentation().runOnMainSync {
-      run {
-        currentActivity = ActivityLifecycleMonitorRegistry.getInstance().getActivitiesInStage(
-          Stage.RESUMED
-        ).elementAtOrNull(0)
-      }
-    }
-    return currentActivity
-  }
-
-  private inline fun <reified T : Activity> isVisible(): Boolean {
-    val am =
-      InstrumentationRegistry.getInstrumentation().targetContext.getSystemService(
-        ACTIVITY_SERVICE
-      ) as ActivityManager
-    val visibleActivityName = this.getCurrentActivity()!!::class.java.name
-    return visibleActivityName == T::class.java.name
-  }
-
-  private inline fun <reified T : Activity> waitUntilActivityVisible() {
-    val startTime = System.currentTimeMillis()
-    while (!isVisible<T>()) {
-      Thread.sleep(CONDITION_CHECK_INTERVAL)
-      if (System.currentTimeMillis() - startTime >= TIMEOUT) {
-        throw AssertionError(
-          "Activity ${T::class.java.simpleName} not visible after $TIMEOUT milliseconds"
-        )
-      }
-    }
-  }
-
   private fun createProfileChooserActivityIntent(): Intent {
     return ProfileChooserActivity
       .createProfileChooserActivity(ApplicationProvider.getApplicationContext())
   }
 
-  private fun getResources(): Resources {
-    return ApplicationProvider.getApplicationContext<Context>().resources
+  private fun scrollToPosition(position: Int) {
+    onView(withId(R.id.profile_recycler_view)).perform(
+      scrollToPosition<RecyclerView.ViewHolder>(
+        position
+      )
+    )
+  }
+
+  private fun matchStringOnProfileListItem(position: Int, targetView: Int, stringToMatch: String) {
```

Comment:
Suggest verify*... here as well.

Reply:
Adding in https://github.com/oppia/oppia-android/issues/2208

Reply:
Resolving on behalf of Ben